go-controller/pkg/cni/cni.go

```diff
@@ -174,9 +174,26 @@ func (pr *PodRequest) cmdAddWithGetCNIResultFunc(kubeAuth *KubeAPIAuth, clientse
 			if err != nil {
 				return nil, err
 			}
-			if _, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo); err != nil {
+			primaryUDNResult, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo)
+			if err != nil {
 				return nil, err
 			}
+
+			originalNumberOfIfaces := len(response.Result.Interfaces)
+			response.Result.Interfaces = append(response.Result.Interfaces, primaryUDNResult.Interfaces...)
+			for i := range primaryUDNResult.IPs {
+				ifaceIPConfig := primaryUDNResult.IPs[i].Copy()
+
+				if ifaceIPConfig.Interface != nil {
+					newIDX := *ifaceIPConfig.Interface + originalNumberOfIfaces
+					ifaceIPConfig.Interface = &newIDX
+				}
+				response.Result.IPs = append(response.Result.IPs, ifaceIPConfig)
+			}
+
+			if len(primaryUDNResult.Routes) >= 0 {
```

Comment:
I don't think the len check is actually needed

Reply:
Done.